app/models/queue_item.rb

```diff
@@ -12,11 +14,27 @@ class QueueItem < ActiveRecord::Base
   delegate :title, to: :video, prefix: :video
   
   def rating
-    review = Review.where(user_id: user.id, video_id: video.id).first
     review.rating if review
   end
   
+  def rating=(new_rating)
+    if review && new_rating.blank?
+      review.destroy
```

Comment:
This really depends on your business logic here. So cannot say this approach is not right. So it all depends. If this is what works for your app then cool :)